ui/src/logs/utils/matchSections.ts

```diff
@@ -0,0 +1,52 @@
+import {MatchType, Filter, MatchSection} from 'src/types/logs'
+import uuid from 'uuid'
+
+export const getMatchSections = (
+  filters: Filter[],
+  text: string
+): MatchSection[] => {
+  if (filters.length === 0) {
+    return [createSection(MatchType.NONE, text)]
+  }
+
+  try {
+    const pattern = filtersToPattern(filters)
+    return sectionOnPattern(pattern, text)
+  } catch (e) {
+    console.error('Syntax Error: bad search filter expression')
+
+    return [createSection(MatchType.NONE, text)]
+  }
+}
+
+const sectionOnPattern = (pattern: RegExp, text) => {
```

Comment:
1. not clear on the surface what this function will do. is this really `matchSections`?
1. maybe `re` instead of `pattern`?